langchain/src/stores/message/utils.ts

```diff
@@ -6,29 +6,64 @@ import {
   SystemChatMessage,
 } from "../../schema/index.js";
 
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export type AdditionalKwargs = Record<string, any>;
+export interface StoredMessageData {
+  content: string;
+  role: string | undefined;
+  additional_kwargs?: AdditionalKwargs;
+}
+
 export interface StoredMessage {
+  type: string;
+  data: StoredMessageData;
+}
+
+interface StoredMessageV1 {
   type: string;
   role: string | undefined;
   text: string;
 }
 
+export function mapV1MessageToStoredMessage(
+  message: StoredMessage | StoredMessageV1
+): StoredMessage {
+  if ("data" in message) {
```

Comment:
Might be some weird edge case where message has the `data` property set but is `undefined` - the more idiomatic way to do this is (weirdly):
```
if (message.data !== undefined) {
}
```

Reply:
Cool thank you!